### ISSUE TITLE
With Typescript, type intersection for onChange with SVGProps  

### ISSUE THREAD

**Issue by raomin** [recharts/recharts#2480]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/fervent-noether-vd5qb)

### Steps to reproduce
Implement onChange method for Brush in a Typescript environment.
```ts
function App() {
  var brushChange = (newIndex: BrushStartEndIndex) => {
    console.log(newIndex);
  };
  return (
    <LineChart
      width={600}
      height={300}
      data={data}
      margin={{ top: 5, right: 30, left: 20, bottom: 5 }}
    >
      <Brush onChange={brushChange} />
      <XAxis dataKey="name" />
...
```

### What is expected?
This code should work.

### What is actually happening?
Error in compiling
```
Type '(newIndex: BrushStartEndIndex) => void' is not assignable to type '((event: FormEvent<SVGElement>) => void) & ((newIndex: BrushStartEndIndex) => void)'.
```

The Brush props is defined as 
```ts
export declare type Props = SVGProps<SVGElement> & BrushProps;
```
BrushProps defines `onChange` but SVGProps also defines `onChange`.
So the resulting Props is an intersecting onChange definition:
```ts
(JSX attribute) onChange?: (((event: React.FormEvent<SVGElement>) => void) & ((newIndex: BrushStartEndIndex) => void)) | undefined
```

There might be a way to specify only one type in the implementation, but I don't know it...

Omitting the SVGProps `onChange` does fix the issue (but might prevent legitimate use of it).

```ts
export declare type Props = Omit<SVGProps<SVGElement>,'onChange'> & BrushProps;
```

NB: It could be useful to export BrushStartEndIndex too!

| Environment | Info |
|---|---|
| Recharts | v2.0.8 |
| React | 17.0.1 |
| System | nodejs 6.14.11 |
| Browser | Chrome 86.0.4240 |

**Comment by mositori** [comment]
I have same issue and it seems nothing has been done for it.
I'd like to tackle this issue if it's okay.

**Comment by sea-dhassouni** [comment]
+1

**Comment by aikastor** [comment]
+ 1

**Comment by aikastor** [comment]
@raomin can you please provide an example of how to `achieve Omitting the SVGProps onChange`? where should I put it?

**Comment by raomin** [comment]
You should declare the new Props anywhere in your tsx
```c++
 export declare type Props = Omit<SVGProps<SVGElement>,'onChange'> & BrushProps;
```

**Comment by natBizitza** [comment]
> ```c++
>  export declare type Props = Omit<SVGProps<SVGElement>,'onChange'> & BrushProps;
> ```

Where to pass that _Props_ then? Thank you!

**Comment by natBizitza** [comment]
Is it solved already? Recharts v2.1.2

**Comment by natBizitza** [comment]
OK, made it work with _any_ 😬 

` const onChange = (e: any) => {
    setStartIndex(e.startIndex);
    setEndIndex(e.endIndex);
  };`

**Comment by ckifer** [comment]
Closed with merging of #3129 

Thanks @montoyaaguirre !
